src/Number.php

```diff
@@ -58,18 +57,8 @@ public static function fromString(string $number): self
     }
 
     /** @psalm-pure */
-    public static function fromFloat(float $number): self
+    public static function fromNumber(int|string $number): self
     {
-        return self::fromString(sprintf('%.14F', $number));
```

Comment:
While I know you marked this class as internal, and that we are breaking BC, I simply know from issue history that  people are relying on this. I don't think I find removing `Number::fromFloat` completely necessary per se.

Reply:
I can restore it, and we can avoid using it in the code: not much to revert - will do, but please note that `Number` is now `@internal`, and people shouldn't rely on it anymore

Reply:
Please do. I understand your remark, and I will take it into account when writing upgrade docs.

Reply:
0ca293324ac1033a1357bae564445f3dddac7cea

Reply:
756649c07e2bb827e08bcc9b09b3f880d7b0fb73